app/src/main/java/pl/szczodrzynski/edziennik/ui/dialogs/timetable/GenerateBlockTimetableDialog.kt

```diff
@@ -373,25 +376,31 @@ class GenerateBlockTimetableDialog(
 
             val today = Date.getToday().stringY_m_d
             val now = Time.getNow().stringH_M_S
-
-            val outputDir = Environment.getExternalStoragePublicDirectory("Szkolny.eu").apply { mkdirs() }
-            val outputFile = File(outputDir, "plan_lekcji_${app.profile.name}_${today}_${now}.png")
+            val filename = "plan_lekcji_${app.profile.name}_${today}_${now}.png"
+            val resolver: ContentResolver = activity.applicationContext.contentResolver
+            val values = ContentValues()
+            values.put(MediaStore.MediaColumns.MIME_TYPE, "image/png")
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                values.put(MediaStore.MediaColumns.DISPLAY_NAME, filename)
+                values.put(MediaStore.MediaColumns.RELATIVE_PATH, Environment.DIRECTORY_PICTURES + File.separator + "Szkolny.eu")
+            } else {
+                val picturesDirectory = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES).absolutePath + File.separator + "Szkolny.eu"
```

Comment:
A cleaner solution would be to work on the File, not on a bare path. The `File()` constructor provides a way to obtain a child file/directory, so you can provide the `Environment...` and "Szkolny.eu" string as its params.